schema/config.schema.json

```diff
@@ -117,31 +117,71 @@
                 },
                 "integration_type": {
                     "type": "string",
-                    "enum": [ 
+                    "enum": [
+                        "oauth_server_to_server",
+                        "oauth_server_to_server_migrate",
                         "oauthweb", 
                         "oauthios", 
                         "oauthandroid", 
                         "service", 
                         "apikey", 
                         "oauthwebapp", 
                         "oauthnativeapp", 
-                        "oauthsinglepageapp",
-                        "oauth_server_to_server"
+                        "oauthsinglepageapp"
                     ]
                 },
+                "oauth_server_to_server": { "$ref": "#/definitions/oauth_server_to_server" },
                 "jwt": { "$ref": "#/definitions/jwt" },
                 "api_key": { "$ref": "#/definitions/api_key" }
             },
             "required": [ "id", "name" ],
-            "if": { "properties": { "integration_type": { "const": "oauthsinglepageapp" } } },
-            "then": { "properties": { "oauth2": { "$ref": "#/definitions/oauthsinglepageapp" } } },
-            "else": { "properties": { "oauth2": { "$ref": "#/definitions/oauth2" } } },
+            "allOf": [
+                {
+                  "if": { "properties": { "integration_type": { "const": "oauthsinglepageapp" } } },
+                  "then": { "properties": { "oauth2": { "$ref": "#/definitions/oauthsinglepageapp" } } },
+                  "else": { "properties": { "oauth2": { "$ref": "#/definitions/oauth2" } } }
+                }
+            ],
             "oneOf": [
-                { "required": ["oauth2"] },
-                { "required": ["jwt"] },
-                { "required": ["api_key"] },
-                { "required": ["oauth_server_to_server"] }
-
+                {
+                  "required": ["oauth2"],
+                  "allOf": [
+                    { "not": { "required": ["api_key"] } },
+                    { "not": { "required": ["jwt"] } },
+                    { "not": { "required": ["oauth_server_to_server"] } }
+                  ] 
+                },
+                {
+                  "required": ["jwt"],
+                  "allOf": [
+                    { "not": { "required": ["api_key"] } },
+                    { "not": { "required": ["oauth2"] } },
+                    { "not": { "required": ["oauth_server_to_server"] } }
+                  ] 
+                },
+                {
+                  "required": ["api_key"],
+                  "allOf": [
+                    { "not": { "required": ["jwt"] } },
+                    { "not": { "required": ["oauth2"] } },
+                    { "not": { "required": ["oauth_server_to_server"] } }
+                  ] 
+                },
+                {
+                  "required": ["oauth_server_to_server"],
```

Comment:
just asking, have you tried to do a `"not": { "required": ["oauth_server_to_server_migrate", "service] }` here for the mutually exclusive check?
See https://stackoverflow.com/questions/28162509/mutually-exclusive-property-groups

Reply:
ok sounds good :)

Reply:
For posterity: These checks are valid... If you add an oauth credential in the console, and then `aio app add service` with the cli, it will create a jwt credential in your workspace, and you will still have the oauth credential you made manually in the workspace. Hence you will have both `service` and `oauth_server_to_server`... 

Still going to refrain from trying new schema stuff as it currently works as is 👍

Reply:
okie, thanks for checking!